overlord/servicestate/quota_control.go

```diff
@@ -100,6 +177,16 @@ func quotaGroupsAvailable(st *state.State) error {
 	if !enableQuotaGroups {
 		return fmt.Errorf("experimental feature disabled - test it by setting 'experimental.quota-groups' to true")
 	}
+
+	// check if the systemd version is too old
+	systemdVersion, err := systemd.Version()
```

Comment:
Should we cache version info? Not super critical, but it seems a bit excessive to query version info on every quota op?

Reply:
Maybe just have a package-level flag initialized in init()? I guess any systemd upgrade would mean eventual snapd restart?

Reply:
> I guess any systemd upgrade would mean eventual snapd restart?  

that is unclear, otoh it's probably very rare that we get downgrades that remove a feature also at that point we are out of luck anyway if the feature was already in use.

Basically it seems that if we had a systemd.AtLeast(version)  bool  (that means it panics if we can't run systemctl ?) we could use it indeed at the top level

Reply:
I have this cached now with a helper to mock it for things that really need to test it